pkgs/development/beam-modules/build-mix.nix

```diff
@@ -8,12 +8,20 @@
 , beamDeps ? []
 , postPatch ? ""
 , compilePorts ? false
+, installPhase ? null
+, buildPhase ? null
+, configurePhase ? null
 , meta ? {}
+, enableDebugInfo ? false
 , ... }@attrs:
 
 with stdenv.lib;
 
 let
+  debugInfoFlag = if enableDebugInfo || elixir.debugInfo
+  then "--debug-info"
+  else "";
```

Comment:
you can also use `lib.optionalString (enableDebugInfo || elixir.debugInfo) "--debug-info";`